src/robotFactory.js

```diff
@@ -1,10 +1,88 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
+  }
 
-class DeliveryDrone {}
+  getInfo() {
+    const { name, weight, chipVersion } = this;
+
+    return `Robot: ${name}, Chip version:`
+      + ` ${chipVersion}, Weight: ${weight}`;
+  };
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, { x, y, z = 0 }, chipVersion) {
+    super(name, weight, {
+      x, y,
+    }, chipVersion);
```

Comment:
it turned out a little wrong after correcting the previous comment, you now have a mixed stylization of the parameters: half of them are located in a column, and half in a row, so not good.
or go back to the style comment from [SerhiiKirik](https://github.com/mate-academy/js_robot_factory/pull/1703#discussion_r1163758858)  or put everything in one line

```suggestion
  constructor(name, weight, coords, chipVersion) {
    super(name, weight, coords, chipVersion);
    
    const { x, y, z = 0 } = coords;
```